libraries/chain/CMakeLists.txt

```diff
@@ -129,8 +129,17 @@ add_library( eosio_chain
              ${HEADERS}
              )
 
+## Boost::accumulators depends on Boost::numeric_ublas, which is still missing cmake support (see
+## https://github.com/boostorg/cmake/issues/39). Until this is fixed, manually add Boost::numeric_ublas
+## as an interface library
+## ----------------------------------------------------------------------------------------------------
+add_library(boost_numeric_ublas INTERFACE)
+add_library(Boost::numeric_ublas ALIAS boost_numeric_ublas)
+
 target_link_libraries( eosio_chain PUBLIC bn256 fc chainbase eosio_rapidjson Logging IR WAST WASM
                        softfloat builtins ${CHAIN_EOSVM_LIBRARIES} ${LLVM_LIBS} ${CHAIN_RT_LINKAGE}
+                       Boost::signals2 Boost::hana Boost::property_tree Boost::multi_index Boost::asio Boost::lockfree
+                       Boost::assign Boost::accumulators
```

Comment:
Having to manually specify every minute boost library like this is a real bummer. I was under the impression 1.82 was going to resolve this, but I guess not?

It feels high maintenance and fragile: what happens when a developer misses that they need to add `Boost::json` (or whatever) to their cmake dependency list? I guess it would be an error if the user doesn't have a system boost. But if a system boost is installed, it could use that _hopefully_ leading to a compile error due to the mismatch; but potentially it could be worse. Should we remove
https://github.com/AntelopeIO/leap/blob/0ae0289621f0f5a528aabdcdf739fd91a5ec8651/.cicd/platforms/ubuntu20.Dockerfile#L9
so we can detect errors with more certainty?

I wonder if we should create our own `Boost::headers` target in some central location. Possibly populating it automatically, or even manually?

Reply:
If you forget to add a library, leap should not build, so I don't quite see how this makes it more fragile. On the other hand I think it is nice to have the dependencies properly listed.

Reply:
> If you forget to add a library, leap should not build

Can't be guaranteed afaict. If you forget `Boost::json` then `#include <boost/json.hpp>` will include the system's boost (provided it's installed). This can lead to mixed versions of boost being used which will hopefully be a compilation failure.. but that isn't guaranteed.

Reply:
Hum, you're right. I guess I was misled because I had to add those dependencies in order to build, because the boost ubuntu package is *not* installed on my system.

Reply:
We could run the CI on systems without boost installed, in which case leap would not build if a boost dependency is missing. And we would always know what we depend on.

Reply:
Right, that's why I was thinking we remove
https://github.com/AntelopeIO/leap/blob/0ae0289621f0f5a528aabdcdf739fd91a5ec8651/.cicd/platforms/ubuntu20.Dockerfile#L9
and
https://github.com/AntelopeIO/leap/blob/0ae0289621f0f5a528aabdcdf739fd91a5ec8651/.cicd/platforms/ubuntu22.Dockerfile#L9

We should also remove the boost build from the pinned build script

Reply:
Thanks Matt, I'll make the changes.

Reply:
I just tried the pinned build. After cloning leap and switching to the `boost_submodule` branch (which add the boost submodule), you have to run `git submodule update --init --recursive` to populate it.

Thant being said, I tested the pinned build and it worked fine.